Working on Linux, but failing on Darwin

```
m 31s
2s
2s
40s
2m 15s
2m 25s
/nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include/c++/v1/__functional/binary_function.h:49:1: note: '__binary_function' declared here
using __binary_function = __binary_function_keep_layout_base<_Arg1, _Arg2, _Result>;
^
In file included from ChemmineOB.cpp:1581:
/nix/store/rcz5spinmx8j2cm3l4kc6qlbi7iijvla-openbabel-unstable-06-12-23/include/openbabel3/openbabel/bond.h:58:15: warning: 'Visit' is deprecated: Use OBBitVec objects instead to be fully thread-safe. [-Wdeprecated-declarations]
  class OBAPI OBBond: public OBBase
              ^
/nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include/c++/v1/__memory/allocator.h:168:28: note: in implicit copy constructor for 'OpenBabel::OBBond' first required here
        ::new ((void*)__p) _Up(_VSTD::forward<_Args>(__args)...);
                           ^
/nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include/c++/v1/__memory/allocator_traits.h:296:13: note: in instantiation of function template specialization 'std::allocator<OpenBabel::OBBond>::construct<OpenBabel::OBBond, const OpenBabel::OBBond &>' requested here
        __a.construct(__p, _VSTD::forward<_Args>(__args)...);
            ^
/nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include/c++/v1/__memory/uninitialized_algorithms.h:553:31: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<OpenBabel::OBBond>>::construct<OpenBabel::OBBond, const OpenBabel::OBBond &, void>' requested here
    allocator_traits<_Alloc>::construct(__alloc, std::__to_address(__first2), *__first1);
                              ^
/nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include/c++/v1/vector:1029:22: note: in instantiation of function template specialization 'std::__uninitialized_allocator_copy<std::allocator<OpenBabel::OBBond>, std::__wrap_iter<const OpenBabel::OBBond *>, std::__wrap_iter<const OpenBabel::OBBond *>, OpenBabel::OBBond *>' requested here
  __tx.__pos_ = std::__uninitialized_allocator_copy(__alloc(), __first, __last, __tx.__pos_);
                     ^
/nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include/c++/v1/vector:1159:9: note: in instantiation of function template specialization 'std::vector<OpenBabel::OBBond>::__construct_at_end<std::__wrap_iter<const OpenBabel::OBBond *>, 0>' requested here
        __construct_at_end(__first, __last, __n);
        ^
ChemmineOB.cpp:2219:18: note: in instantiation of function template specialization 'std::vector<OpenBabel::OBBond>::vector<std::__wrap_iter<const OpenBabel::OBBond *>, 0>' requested here
      return new Sequence(vb, ve);
                 ^
ChemmineOB.cpp:3481:20: note: in instantiation of function template specialization 'swig::getslice<std::vector<OpenBabel::OBBond>, long>' requested here
      return swig::getslice(self, i, j);
                   ^
/nix/store/rcz5spinmx8j2cm3l4kc6qlbi7iijvla-openbabel-unstable-06-12-23/include/openbabel3/openbabel/bond.h:97:7: note: 'Visit' has been explicitly marked deprecated here
      OB_DEPRECATED_MSG("Use OBBitVec objects instead to be fully thread-safe.")
      ^
/nix/store/rcz5spinmx8j2cm3l4kc6qlbi7iijvla-openbabel-unstable-06-12-23/include/openbabel3/openbabel/babelconfig.h:38:38: note: expanded from macro 'OB_DEPRECATED_MSG'
    #define OB_DEPRECATED_MSG(msg) [[deprecated(msg)]]
                                     ^
46 warnings and 7 errors generated.
make: *** [/nix/store/zaz1hm676fmxcr131i6g0jqm4c00f8sk-R-4.4.1/lib/R/etc/Makeconf:202: ChemmineOB.o] Error 1
ERROR: compilation failed for package ‘ChemmineOB’
* removing ‘/nix/store/58i28j92pz11a212vjmxbcqsxz7lkn48-r-ChemmineOB-1.42.0/library/ChemmineOB’
error: builder for '/nix/store/6kv1fx3b6ywwyhgd7la1vki1gc75hj70-r-ChemmineOB-1.42.0.drv' failed with exit code 1;
       last 25 log lines:
       > /nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include/c++/v1/__memory/uninitialized_algorithms.h:553:31: note: in instantiation of function template specialization 'std::allocator_traits<std::allocator<OpenBabel::OBBond>>::construct<OpenBabel::OBBond, const OpenBabel::OBBond &, void>' requested here
       >     allocator_traits<_Alloc>::construct(__alloc, std::__to_address(__first2), *__first1);
       >                               ^
       > /nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include/c++/v1/vector:1029:22: note: in instantiation of function template specialization 'std::__uninitialized_allocator_copy<std::allocator<OpenBabel::OBBond>, std::__wrap_iter<const OpenBabel::OBBond *>, std::__wrap_iter<const OpenBabel::OBBond *>, OpenBabel::OBBond *>' requested here
       >   __tx.__pos_ = std::__uninitialized_allocator_copy(__alloc(), __first, __last, __tx.__pos_);
       >                      ^
       > /nix/store/z6r3bn5l51679pwkvh9nalp6c317z34m-libcxx-16.0.6-dev/include/c++/v1/vector:1159:9: note: in instantiation of function template specialization 'std::vector<OpenBabel::OBBond>::__construct_at_end<std::__wrap_iter<const OpenBabel::OBBond *>, 0>' requested here
       >         __construct_at_end(__first, __last, __n);
       >         ^
       > ChemmineOB.cpp:2219:18: note: in instantiation of function template specialization 'std::vector<OpenBabel::OBBond>::vector<std::__wrap_iter<const OpenBabel::OBBond *>, 0>' requested here
       >       return new Sequence(vb, ve);
       >                  ^
       > ChemmineOB.cpp:3481:20: note: in instantiation of function template specialization 'swig::getslice<std::vector<OpenBabel::OBBond>, long>' requested here
       >       return swig::getslice(self, i, j);
       >                    ^
       > /nix/store/rcz5spinmx8j2cm3l4kc6qlbi7iijvla-openbabel-unstable-06-12-23/include/openbabel3/openbabel/bond.h:97:7: note: 'Visit' has been explicitly marked deprecated here
       >       OB_DEPRECATED_MSG("Use OBBitVec objects instead to be fully thread-safe.")
       >       ^
       > /nix/store/rcz5spinmx8j2cm3l4kc6qlbi7iijvla-openbabel-unstable-06-12-23/include/openbabel3/openbabel/babelconfig.h:38:38: note: expanded from macro 'OB_DEPRECATED_MSG'
       >     #define OB_DEPRECATED_MSG(msg) [[deprecated(msg)]]
       >                                      ^
       > 46 warnings and 7 errors generated.
       > make: *** [/nix/store/zaz1hm676fmxcr131i6g0jqm4c00f8sk-R-4.4.1/lib/R/etc/Makeconf:202: ChemmineOB.o] Error 1
       > ERROR: compilation failed for package ‘ChemmineOB’
       > * removing ‘/nix/store/58i28j92pz11a212vjmxbcqsxz7lkn48-r-ChemmineOB-1.42.0/library/ChemmineOB’
       For full logs, run 'nix log /nix/store/6kv1fx3b6ywwyhgd7la1vki1gc75hj70-r-ChemmineOB-1.42.0.drv'.
error: 1 dependencies of derivation '/nix/store/xrbizm7m81wg4zwm7i6v92jd92m9lzp1-nix-shell.drv' failed to build 
```

--- a/pkgs/development/r-modules/default.nix
+++ b/pkgs/development/r-modules/default.nix
@@ -801,7 +801,7 @@ let
     sphereTessellation = with pkgs; [ gmp.dev mpfr.dev ];
     vapour = with pkgs; [ proj.dev gdal ];
     MedianaDesigner = [ pkgs.zlib.dev ];
-    ChemmineOB = [ pkgs.eigen ];
+    ChemmineOB = with pkgs; [ eigen openbabel ];
     DGP4LCF = [ pkgs.lapack pkgs.blas ];
   };
 
@@ -1582,29 +1582,26 @@ let
       PKGCONFIG_LIBS = "-Wl,-rpath,${lib.getLib pkgs.openssl}/lib -L${lib.getLib pkgs.openssl}/lib -L${pkgs.cyrus_sasl.out}/lib -L${pkgs.zlib.out}/lib -lssl -lcrypto -lsasl2 -lz";
     });
 
-    ChemmineOB = let
-      # R package doesn't compile with the latest (unstable) version.
-      # Override from nixpkgs-23.11
-      openbabel3 = pkgs.openbabel.overrideAttrs (attrs: {
-        version = "3.1.1";
-        src = pkgs.fetchFromGitHub {
-          owner = "openbabel";
-          repo = "openbabel";
-          rev = "openbabel-${lib.replaceStrings ["."] ["-"] attrs.version}";
-          sha256 = "sha256-wQpgdfCyBAoh4pmj9j7wPTlMtraJ62w/EShxi/olVMY=";
-        };
-      });
-    in
-    old.ChemmineOB.overrideAttrs (attrs: {
+    ChemmineOB = old.ChemmineOB.overrideAttrs (attrs: {
       # pkg-config knows openbabel-3 without the .0
       # Eigen3 is also looked for in the wrong location
+      # pointer was changed in newer version of openbabel:
+      #   https://github.com/openbabel/openbabel/commit/305a6fd3183540e4a8ae1d79d10bf1860e6aa373
       postPatch = ''
         substituteInPlace configure \
           --replace-fail openbabel-3.0 openbabel-3
         substituteInPlace src/Makevars.in \
           --replace-fail "-I/usr/include/eigen3" "-I${pkgs.eigen}/include/eigen3"
+        substituteInPlace src/ChemmineOB.cpp \
+          --replace-fail "obsharedptr<" "std::shared_ptr<"
       '';
-      buildInputs = attrs.buildInputs ++ [openbabel3];
+
+      # copied from fastnlo-toolkit:
+      # None of our currently packaged versions of swig are C++17-friendly
+      # Use a workaround from https://github.com/swig/swig/issues/1538
+      env = (attrs.env or { }) // {
+        NIX_CFLAGS_COMPILE = (attrs.env.NIX_CFLAGS_COMPILE or "") + lib.optionalString stdenv.hostPlatform.isDarwin " -D_LIBCPP_ENABLE_CXX17_REMOVED_FEATURES";
+      };
     });
 
     ps = old.ps.overrideAttrs (attrs: {

